x-pack/plugins/actions/server/actions_client/actions_client.ts

```diff
@@ -727,6 +727,21 @@ export class ActionsClient {
       trackLegacyRBACExemption('execute', this.context.usageCounter);
     }
 
+    // we need to throw the actionType not found/enabled error AFTER the authorization check
+    if (!actionTypeId) {
+      if (typeNotFoundErr) throw typeNotFoundErr;
+      // defensive, in case of future code changes and typeNotFoundErr isn't set
+      throw new Error(`Action type not found for actionId: ${actionId}`);
+    }
+
+    try {
+      this.context.actionTypeRegistry.ensureActionTypeEnabled(actionTypeId);
+    } catch (err) {
+      const message = `Attempting to execute a disabled action type ${err}`;
+      this.context.logger.warn(message);
+      return { actionId, status: 'error', message };
```

Comment:
Seems like the change to `actionTypeRegistry.isActionExecutable` should be sufficient, do we need this extra check here?

Reply:
Ya, I want to try that out.  The `isActionExecutable()` code was needed for action-based execution, because it doesn't go through the action client :-), which I realized after adding this code.  I haven't tried pulling this out to see if this fixes http as well.  Will try it today ...

Reply:
I noticed that too! The full error message is in the response body in a different field but it's not being shown in the UI.